tools/relay-cache-warmer/graphql.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const query = `query TwinsQuery {
+  items: twinsConnection(first: %d, orderBy: id_ASC%s) {
+    pageInfo {
+  	  endCursor
+  	  hasNextPage
+    }
+  }
+  twins(limit: %d, offset: %s, orderBy: id_ASC) {
+    accountID
+    publicKey
+    twinID
+    relay
+  }
+}`
+
+const twinsPerPage = 500
+
+type Twin struct {
+	ID        uint      `json:"id"`
+	AccountID string    `json:"account"`
+	Relay     []string  `json:"relay"`
+	PK        JSONArray `json:"pk"`
+}
+
+// only exists because json marshals []byte to a string instead of json array
+type JSONArray []byte
+
+func (u JSONArray) MarshalJSON() ([]byte, error) {
+	if u == nil || len(u) == 0 {
+		return []byte("null"), nil
+	}
+
+	var buf bytes.Buffer
+	_, err := buf.WriteRune('[')
+	if err != nil {
+		return nil, err
+	}
```

Comment:
i think the buffer write should never fail (u can safely ignore the error) unless it failed to allocate memory which means your entire app will crash anyway.

If you check implementation for [WriteRune](https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/bytes/buffer.go;l=299) or even `Write` you will see it never returns an error

So in your code u can ignore the error or do something like 
```go
_, _ = buf.WriteRune('[') 
```
to make sure linters are happy